### ISSUE TITLE
Remove duplicate `fs.readFileSync` call

### ISSUE THREAD

**Issue by andy-hanson** [palantir/tslint#2584]
In `runner.ts`, we do:

```ts
const contents = fs.readFileSync(file, "utf8");
...
linter.lint(file, contents, configFile);
```

Then in `linter.ts`, we do:

```ts
let sourceFile = this.getSourceFile(fileName, source);
```

and in `getSourceFile` in the same file:
```ts
private getSourceFile(fileName: string, source: string) {
    if (this.program) {
        ... don't use source ...
    } else { ... }
}
```

So passing in the source text is unnecessary for a type-checked lint. The API should provide a way of asking for failures for a file without passing in the source text.
